src/main/java/me/jellysquid/mods/lithium/mixin/entity/gravity_check_block_below/MixinVoxelShapes.java

```diff
@@ -0,0 +1,43 @@
+package me.jellysquid.mods.lithium.mixin.entity.gravity_check_block_below;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.block.ShapeContext;
+import net.minecraft.util.math.*;
+import net.minecraft.util.shape.VoxelShape;
+import net.minecraft.util.shape.VoxelShapes;
+import net.minecraft.world.WorldView;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+import java.util.stream.Stream;
+
+@Mixin(VoxelShapes.class)
+public class MixinVoxelShapes {
+    /**
+     * Check the block below the entity first, as it is the block that is most likely going to cancel the movement from
+     * gravity.
+     */
+    @Inject(method = "calculatePushVelocity(Lnet/minecraft/util/math/Box;Lnet/minecraft/world/WorldView;DLnet/minecraft/block/ShapeContext;Lnet/minecraft/util/math/AxisCycleDirection;Ljava/util/stream/Stream;)D",
+            at = @At(value = "INVOKE", target = "Lnet/minecraft/util/math/AxisCycleDirection;opposite()Lnet/minecraft/util/math/AxisCycleDirection;", ordinal = 0),
+            cancellable = true, locals = LocalCapture.NO_CAPTURE)
+    private static void checkBelowFeet(Box box, WorldView world, double initial, ShapeContext context, AxisCycleDirection direction, Stream<VoxelShape> shapes, CallbackInfoReturnable<Double> cir) {
+        //                  [VanillaCopy]
```

Comment:
These `[VanillaCopy]` annotations do not explain what vanilla behavior they are copying and defeat the purpose of using them. Additional documentation on what exactly is going on here should be included.

Reply:
Okay I added a more detailed description now